### PR TITLE
docs(transform): document the fifth workflow, and correct what the fourth made false

### PR DESCRIPTION
**Pull request by vboussot**
Base : #79.

Une page de configuration pour `Transformer:` — le plan et ses verdicts, `Save` contre `Write`, les refus au parse, `Reduce`, `Expand`, et comment écrire un transform qui streame.

**Ses exemples sont exécutables.** `tests/integration/test_transform_doc_examples.py` extrait chaque bloc YAML de la page plutôt que de le recopier, et chacun est planifié **et exécuté** — un plan se calcule sur le lanceur, le run spawne ensuite, donc un plan vert ne prouve rien sur ce qui traverse cette frontière.

Le reste corrige ce qu'un cinquième workflow rend **faux** et non seulement incomplet : « run three ways » en page d'accueil, la liste de dispatch d'`execution-flow`, les clés racines de `configuration`, les régimes de `streaming`, la carte des modules de `datasets`, les options « communes » du CLI (TRANSFORM construit son propre parseur et n'a pas de `-tb`), et les listes autodoc, qui ignoraient trois modules. `subset:` était décrit comme un mapping ou un objet sur trois pages — la seule forme que le binder n'accepte pas.

**`Warp` accepte `max_displacement: auto`** et rejoint le catalogue des transforms, où il ne figurait pas. Le halo d'un warp est la distance dont le champ pousse réellement : une propriété de la donnée, pas d'un paramètre. Un champ enregistre sa borne par composante quand KonfAI l'écrit, donc `auto` la relit dans les en-têtes. C'est la borne de la cohorte : elle sur-lit pour un cas doux et n'est jamais courte pour un cas violent ; sans borne lisible, l'étage déclare le volume entier et le dit.

## Summary by CodeRabbit

- **New Features**
  - Added the model-free `TRANSFORM` workflow with CLI support, planning, execution, resumability, and workspace outputs.
  - Added transform-only pipelines with streaming, whole-volume processing, reduction, expansion, and synchronized augmentation.
  - Enhanced `Warp` with automatic displacement-bound discovery, component-specific validation, and anisotropic locality handling.

- **Documentation**
  - Added comprehensive transform configuration, CLI, workflow, API, dataset, and execution-flow documentation.
  - Clarified dataset subset selectors and memory-budget behavior.

- **Tests**
  - Added coverage for documented transform examples, automatic bounds, fallback behavior, validation, and anisotropic halos.

### CODE REVIEW

**Comment by vboussot** [comment]
@coderabbitai review

**Comment by vboussot** [comment]
@coderabbitai review

**Comment by vboussot** [comment]
@coderabbitai review

File: docs/source/concepts/execution-flow.md [hunk 1]
```diff
@@ -19,13 +22,15 @@ The `konfai` CLI dispatches to three public functions:
 - `konfai.trainer.train`
 - `konfai.predictor.predict`
 - `konfai.evaluator.evaluate`
+- `konfai.transformer.transform`
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/concepts/execution-flow.md:25]: Confirmed and fixed for the first location: the line said three and the list below it has four. The other two locations are not applied — `config_guide/index.rst` disambiguates "the other two" in the next clause and scopes the fourth explicitly, and `index.rst` already reads "One config, built once, run four ways" with all four lanes in the table.

File: docs/source/concepts/streaming.md [hunk 2]
```diff
@@ -36,7 +36,8 @@ per augmented copy, not once per dataset, so a chain that streams for one
 augmentation draw may load the volume for the next.
 
 The regime is not a config key. Training defaults to the cache (epochs re-read
-every case); prediction and evaluation default to streaming (one pass). A
+every case); prediction, evaluation and transform default to streaming (one
+pass). A
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/concepts/streaming.md:40]: Not applied. "The three regimes" is the dataset-residency taxonomy (cache / stream / buffer) for the DataLoader workflows, which TRANSFORM does not participate in — a different axis from the plan's verdicts. Both transform facts named here are already on the page, at the paragraph above and the one below.

File: docs/source/config_guide/prediction.md [hunk 3]
```diff
@@ -80,7 +80,7 @@ Key fields:
 | `groups_src` | mapping | Input groups and preprocessing transforms. |
 | `augmentations` | mapping | Test-time augmentation definitions. |
 | `Patch` | mapping | Sliding-window or slice-wise inference setup. |
-| `subset` | object | Restricts evaluated cases. |
+| `subset` | string / list / null | Restricts evaluated cases: a flat selector — a case name, a case-list file, `~file` to exclude, a `start:end` slice, or a list of those. Not a nested mapping. |
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/config_guide/prediction.md]: Confirmed and fixed. The row was copied verbatim from `evaluation.md`, and `konfai EVALUATION` is a separate command with its own config and its own `DataMetric.subset`; it now reads "Restricts which cases are predicted", matching how the training and transform pages phrase their own.

File: docs/source/config_guide/transform.md [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+Read transforms run on CPU; `--gpu` matters only when a chain embeds a
+`KonfAIInference` stage, whose nested inference does use the device. That stage
+is whole-volume and its GPU/RAM usage lives **outside** `memory_budget` — the
+plan prints a note saying it cannot bound those chains. There is no `-tb`: the
+workflow emits no scalars.
+
+## Read the plan before you read anything else
+
+Every run prints its plan first, and writes it to
+`./Transforms/<name>/plan.txt`. Nothing is written until the plan is accepted.
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/config_guide/transform.md:51]: Not applied. The probe writes no output — a `--plan` over an existing destination leaves it byte-for-byte identical — and the one residue it does leave (an empty container) is already spelled out four paragraphs down in the same section: "That is why `--plan` still touches the output directories."

File: docs/source/config_guide/transform.md [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+    worst fallback case ~= 3.10 GiB vs per-rank budget 7.45 GiB
+```
+
+Three verdicts, and each one is a fact about *your* run:
+
+- **STREAM** — the case is read and written region by region. Memory is one
+  slab, whatever the volume's size.
+- **WHOLE-VOLUME** — the case is assembled in memory, then written. Always
+  correct, never bounded. The line says which stage refused and why.
+- **SKIP** — the output already exists; nothing is recomputed.
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/config_guide/transform.md]: Confirmed and fixed. A `Reduce` chain prints `REDUCE` or `REFUSED`, and `REFUSED` refuses the run whatever `on_fallback` says — neither was in the list. The count is gone and both are documented, including why a reduction has no fallback to take.

File: docs/source/config_guide/transform.md [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+        if self.radius == 0:
+            return PatchLocality(LocalityKind.POINTWISE)
+        return PatchLocality(LocalityKind.HALO, halo=(self.radius,))
+
+    def __call__(self, name: str, tensor: torch.Tensor, cache_attribute: Attribute) -> torch.Tensor:
+        if self.radius == 0:
+            return tensor
+        k = 2 * self.radius + 1
+        return F.avg_pool3d(tensor.to(torch.float32), k, stride=1, padding=self.radius).to(tensor.dtype)
+```
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/config_guide/transform.md:351]: Not applied. `Transform.transform_shape()` is a concrete identity default, not an abstract method, and `BoxFilter` is exactly shape-preserving — so the example's inherited prediction is already correct and an override would restate the base class. Marking it required in the table would also contradict `reference/components/transforms.md`, which correctly limits the obligation to transforms that change the spatial shape.

File: docs/source/index.rst [hunk 7]
```diff
@@ -106,11 +106,19 @@ KonfAI
...
+         </div>
          <div class="kf-lfoot">
            <span>Same engine underneath: reflection reads the root key and builds
-                 <code>Model &middot; Dataset &middot; Losses &middot; Optimizer</code>.</span>
-           <span>Every output folder is keyed by <code>train_name</code> &mdash; keep it consistent
-                 across the three files.</span>
+                 <code>Model &middot; Dataset &middot; Losses &middot; Optimizer</code> &mdash;
+                 <code>Transformer:</code> builds a <code>Dataset</code> and nothing else.</span>
+           <span>Every output folder is keyed by <code>train_name</code> (<code>name</code> under
+                 <code>Transformer:</code>) &mdash; keep it consistent across the files.</span>
```
[the remark this answers is not in the document]
Comment (vboussot) [reply docs/source/index.rst:121]: Not applied. The replacement does not remove the ambiguity it names: "wherever each `Write:` says" sits inside the very column headed "Writes to the workspace", so "workspace folder" would be exactly as broad as "output folder" — while borrowing a noun the docs already use for the MCP root, the app-server job directory and the Docker mount. The exception is carried by the row immediately above the sentence.

File: konfai/data/transform.py [hunk 8]
```diff
@@ -1568,9 +1575,55 @@ def __init__(
...
+        bound: list[float] = []
+        for entry in entries:
+            _shape, attribute = self.field_dataset.get_infos(group, entry)
+            if DISPLACEMENT_BOUND_ATTRIBUTE not in attribute:
+                return self._auto_bound
+            recorded = [float(value) for value in attribute.get_np_array(DISPLACEMENT_BOUND_ATTRIBUTE).ravel()]
+            bound = recorded if not bound else [max(a, b) for a, b in zip(bound, recorded, strict=False)]
+        if bound and max(bound) > 0.0:
+            self._auto_bound = bound
+        return self._auto_bound
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/data/transform.py:1626]: Confirmed and fixed — this one was worth the comment. A directory store lists its entries from the filesystem alone, so `get_names` there cannot raise and the guarded region was empty of the failure its own comment names; the per-entry `get_infos` outside it turned one corrupt file anywhere under the field root into a raw SimpleITK error at plan time, for entries the run never warps. The header reads are now inside the guard, with a test that corrupts an entry in a directory-backed field group and asserts the whole-volume answer.

File: tests/integration/test_transform_doc_examples.py [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+@pytest.mark.integration
+@pytest.mark.parametrize("line,config", _doc_examples(), ids=lambda value: value if isinstance(value, int) else "")
+def test_a_documented_example_plans_and_runs(line: int, config: str, tmp_path: Path) -> None:
+    workdir = tmp_path / f"block_{line}"
+    workdir.mkdir()
+    _write_dataset(workdir / "Raw", _source_groups(config))
+    (workdir / "Transform.yml").write_text(config, encoding="utf-8")
+    for filename, source in _python_modules(DOC.read_text(encoding="utf-8")).items():
+        (workdir / filename).write_text(source, encoding="utf-8")
+    environment = {"KONFAI_CLI": "False", "PYTHONPATH": str(workdir)}
```
[the remark this answers is not in the document]
Comment (vboussot) [reply tests/integration/test_transform_doc_examples.py]: Confirmed and fixed for the PYTHONPATH half: the children were importing whatever `konfai` was installed rather than the checkout under test, so green certified another tree. It now composes `harness.subprocess_env()` with the work directory first, like the seven sibling integration tests. The `KONFAI_*` half is not applied — `tests/conftest.py` sets those through `monkeypatch.setenv`, so they are already reverted per test and stripping them would diverge from every other integration test.

File: tests/integration/test_transform_doc_examples.py [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+            image.SetSpacing((0.5, 0.5, 2.0))
+            image.SetOrigin((1.0, 2.0, 3.0))
+            sitk.WriteImage(image, str(case / f"{group}.mha"))
+
+
+def _doc_examples() -> list[tuple[int, str]]:
+    if not DOC.is_file():
+        return []
+    text = DOC.read_text(encoding="utf-8")
+    return [(line, config) for line, body in _yaml_blocks(text) if (config := _as_config(body)) is not None]
```
[the remark this answers is not in the document]
Comment (vboussot) [reply tests/integration/test_transform_doc_examples.py:128]: Not applied. The empty-parametrize state is not reachable here — `transform.md` is tracked, listed in the config-guide toctree, and collects 6 examples today — and when forced it degrades to a visible SKIP, not a green claim of coverage. The proposed patch is also worse than the state it guards: raising at module scope aborts the whole pytest session at collection with a `FileNotFoundError`, not the `AssertionError` it advertises.
